bionic/aip/main.py

```diff
@@ -0,0 +1,70 @@
+"""
+This module is run as main in order to execute a task on a worker.
+"""
+
+import logging
+import os
+import sys
+
+from bionic.deps.optdep import import_optional_dependency
+
+
+def run():
+    # Scope the import to this function to avoid raising for anyone not using it.
+    blocks = import_optional_dependency("blocks", raise_on_missing=True)
+    cloudpickle = import_optional_dependency("cloudpickle", raise_on_missing=True)
+
+    ipath = sys.argv[-1]
+    fs = blocks.filesystem.GCSNativeFileSystem()
+    with fs.open(ipath, "rb") as f:
+        task = cloudpickle.load(f)
+
+    # Now that we have the task, setup logging.
+    _setup_logging(task.job_id(), task.config.project)
+    logging.info(f"Read task from {ipath}")
+
+    logging.info("Starting execution")
+    result = task.function()
+
+    opath = task.output_uri()
+    logging.info(f"Uploading result to {opath}")
+    blocks.pickle(result, opath)
+    return
+
+
+def _setup_logging(job_id, project):
```

Comment:
nit: Should `project` be `project_id`? Likewise for `trial`/`trial_id`?

Reply:
We can rename `trial` to `trial_id`. For `project`, I think the more appropriate name would be `project_name` (instead of id). I'll rename this too.

Reply:
Oh but the google cloud logging `Resource` calls it `project_id`, so I guess I'll just call it `project_id` in this method but `project_name` elsewhere. I'm calling it `project_name` elsewhere because that's what the google API calls it (check this [create API](https://cloud.google.com/ai-platform/training/docs/reference/rest/v1/projects.jobs/create) that takes project).

Reply:
Their naming is all over the place honestly. While making this change, I found out that they also call it `project` in logging `Client`. I'll keep it `project_name`, but again wanted to point out how bad AIP documentation and naming is everywhere.

Reply:
https://www.youtube.com/watch?v=uaehB3PJfr0